File: week_8_and_9/3_Ruby/rectangle_class/my_solution.rb
# U3.W8-9: Implement a Rectangle Class


# I worked on this challenge [by myself, with: ].

# 2. Pseudocode



# 3. Initial Solution

class Rectangle
  attr_accessor :width, :height

  def initialize(width, height)
    @width= width
    @height= height
  end

  def area
    @width * @height
  end

  def perimeter
    (2 * @width) + (2 * @height)
  end

  def diagonal
    sqrt(@width^2 + @height^2)
  end

  def square?
    @width== @height ? true : false
  end


  # def ==(other)
  #   (other.width  == self.width && other.height == self.height ) ||
  #   (other.height == self.width && other.width  == self.height )
  # end

end






# 4. Refactored Solution






# 1. DRIVER TESTS GO BELOW THIS LINE






# 5. Reflection 

File: week_8_and_9/2_JavaScript/grocery_list/my_solution.js
// U3.W8-9: 


// I worked on this challenge with:
// 1) Josh Jeong
// 2) Rj Bernaldo

// 2. Pseudocode
/*
# As a user, I want to add item to grocery list
# As a user, I want to delete item off grocery list
# As a user, I want to see whats included list?
# As a user, I want to see the entire list. show method
# As a user, I want to add item with its quantity
*/

// 3. Ruby code
/*
class GroceryList
    attr_accessor :name, :list
    
    def initialize (name)
        @list = []
    end

    def add(name, price)
      @item = {"name" => name, "price" => price }
      @list << @item
    end
    
    def items
     @list.each  do |item|
       puts "Name: #{item["name"]} Price: #{item["price"]}"
     end
    end
    
    def remove name
      @list.each do |item|
        if item.has_value? name
         @list.delete item
        end
      end
    end    
end

JavaScript Translation:
*/ 

var GroceryList = function(name) {
    this.name = name;
    this.list = [];
    this.add = function(name, price) {
        var item = {
            name: name,
            price: price
        }
        this.list.push(item);
    }
    this.items = function() {
      console.log('items --- ')
        for (i in this.list) {
            sentence = "name: " + this.list[i].name + " price: " + this.list[i].price;
            console.log(sentence);
        }
    }
    this.remove = function(item) {
        for(i in this.list) {
            if (item = this.list[i].name) {
              this.list.splice(i, i)
            }
        }
    }
}

// 4. Refactored Solution

// 1. DRIVER TESTS/ASSERT STATEMENTS GO BELOW THIS LINE
list = new GroceryList('list');
list.items();
list.add("apple", 5)
list.items();
list.add("orange", 3)
list.items();
list.remove("orange")
list.items();


/* 5. Reflection 
Translating the code from one language to another is really helping me understand JavaScript.
This was also my first pairing session in person which made is so muche easier.
Very excited to start school!!!
*/
